client.go

```diff
@@ -31,321 +37,131 @@ func NewClient(apiKey string, options ...Option) *Client {
 		apiKey: apiKey,
 		config: defaultConfig(),
 		client: resty.New(),
+		ctx:    context.Background(),
 	}
 
 	c.client.SetHeaders(map[string]string{
 		"Accept":            "application/json",
 		"X-Environment-Key": c.apiKey,
 	})
+	c.client.SetTimeout(time.Second * c.config.timeout)
 
 	for _, opt := range options {
 		opt(c)
 	}
-
 	if c.config.localEvaluation {
-		go c.pollEnvironment(context.TODO())
+		go c.pollEnvironment(c.ctx)
+	}
+	// Initialize analytics processor
+	if c.config.enableAnalytics {
+		c.analyticsProcessor = NewAnalyticsProcessor(c.ctx, c.client, c.config.baseURL, nil)
 	}
 
 	return c
 }
 
-func (c *Client) SetProxy(proxyUrl string) {
-	c.client.SetProxy(proxyUrl)
-}
-func (c *Client) RemoveProxy() {
-	c.client.RemoveProxy()
-}
-
-// GetFeatures returns all features available in given environment
-func (c *Client) GetFeatures() ([]Flag, error) {
-	return c.GetFeaturesWithContext(context.TODO())
-}
-
-// GetFeaturesWithContext returns all features available in given environment
-func (c *Client) GetFeaturesWithContext(ctx context.Context) ([]Flag, error) {
-	flags := make([]Flag, 0)
-	_, err := c.client.NewRequest().
-		SetContext(ctx).
-		SetResult(&flags).
-		Get(c.config.baseURI + "flags/")
-
-	return flags, err
-}
-
-// GetUserFeatures returns all features as defined for given user
-func (c *Client) GetUserFeatures(user User) ([]Flag, error) {
-	return c.GetUserFeaturesWithContext(context.TODO(), user)
+// Returns `Flags` struct holding all the flags for the current environment
+func (c *Client) GetEnvironmentFlags() (Flags, error) {
+	return c.GetEnvironmentFlagsWithContext(c.ctx)
```

Comment:
Do we need these extra `...WithContext` methods? Can't we just pass the context straight into the e.g. `GetEnvironmentFlagsFromDocument` method?

Reply:
Well, that sort of was the convention in v1, I guess it gives the client a bit more control, I am happy either way tho